cmd/ocitools/validate.go

```diff
@@ -397,6 +398,26 @@ func checkLinuxResources(r rspec.Resources, hostCheck bool) (msgs []string) {
 			msgs = append(msgs, fmt.Sprintf("Minimum memory limit should be larger than memory reservation"))
 		}
 	}
+	if r.Network != nil && hostCheck {
+		var exist bool
+		interfaces, err := net.Interfaces()
+		if err != nil {
+			msgs = append(msgs, err.Error())
+			return
+		}
+		for _, prio := range r.Network.Priorities {
+			exist = false
+			for _, ni := range interfaces {
+				if prio.Name == ni.Name {
+					exist = true
+					break
+				}
+			}
+			if !exist {
+				msgs = append(msgs, fmt.Sprintf("Interface %s does not exist on host", prio.Name))
```

Comment:
On Wed, Aug 24, 2016 at 02:40:39AM -0700, Ma Shimiao wrote:
> runtime uses cgroups to set container's resource limit. I think when
> setting cgroups, we does not need to concern about how namespaces
> have been set.

A few possible positions here:

a. The current spec wording requires errors “if the config specifies
   anything else related to that namespace” [1,2] if a namespace entry
   is listed with a ‘path’ (although see [3]).  I think it's pretty
   clear that network interfaces are related to network namespace
   (each interface belongs to one namespace).  So if we strictly stick
   to the current spec wording, I think we *have* to care about
   namespace setup here.

b. I don't see any problem with configuring a new cgroup which happens
   to be limiting an old namespace, since that won't step on anybody's
   toes (more on this in [3]).  We'd need to change the spec wording
   if folks agree with this.

c. I don't see any problem with configuring an *existing* cgroup
   (assuming you have the appropriate permissions, which the kernel
   will check when you try and change the cgroup).  Folks can setup
   cgroup permissions or audit configs as they see fit to enforce any
   anti-tweaking policy they like, but I don't think we need to bake
   those restrictions into runtime-spec.  We'd need to change the spec
   wording if folks agree with this.

Unless/until we land a spec change (which I'm in favor of), I think we
need to stick to (a).

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc1/config-linux.md#namespaces
[2]: https://github.com/opencontainers/runtime-spec/pull/158
[3]: https://github.com/opencontainers/runtime-spec/issues/537